src/simplify_algebra.cpp

```diff
@@ -760,29 +760,42 @@ struct find_split_reshape
         }
 
         // ensure reshape happens after the axis dimension
-        auto axis    = any_cast<op::slice>(slc->get_operator()).axes[0];
-        auto in_lens = input->get_shape().lens();
-        if(!std::equal(in_lens.begin(), in_lens.begin() + axis, dims.begin(), dims.begin() + axis))
+        auto axis         = any_cast<op::slice>(slc->get_operator()).axes[0];
+        auto slc_lens     = slc->get_shape().lens();
+        auto slc_dim_size = std::accumulate(
+            slc_lens.begin() + axis, slc_lens.end(), 1, std::multiplies<std::size_t>());
+
+        // search the reshape output (standard shape) to decide which axis are
+        // in its output corresponding to the slc_dim_size
+        auto rsp_lens    = rsp->get_shape().lens();
+        auto rsp_strides = rsp->get_shape().strides();
+        rsp_strides.insert(rsp_strides.begin(), rsp_strides[0] * rsp_lens[0]);
+        auto ait = std::find(rsp_strides.begin(), rsp_strides.end(), slc_dim_size);
+        if(ait == rsp_strides.end())
         {
             return;
```

Comment:
Is it possible to write a unit test for this branch?

Reply:
Yes, added one. Thanks.